test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs

```diff
@@ -44,9 +46,17 @@ public partial class GrpcTests
         [InlineData("http://[::1]", false)]
         public void GrpcClientCallsAreCollectedSuccessfully(string baseAddress, bool shouldEnrich = true)
         {
-            var uri = new Uri($"{baseAddress}:{this.server.Port}");
+            var uri = new Uri($"{baseAddress}:1234");
             var uriHostNameType = Uri.CheckHostName(uri.Host);
 
+            var httpClient = ClientTestHelpers.CreateTestClient(async request =>
+            {
+                var streamContent = await ClientTestHelpers.CreateResponseContent(new HelloReply());
+                var response = ResponseUtils.CreateResponse(HttpStatusCode.OK, streamContent, grpcStatusCode: global::Grpc.Core.StatusCode.OK);
+                response.TrailingHeaders().Add("grpc-message", "value");
+                return response;
+            });
```

Comment:
This test uses the pattern used to mock an http client in the `grpc-dotnet` project for testing the `Grpc.Net.Client`

For example:
https://github.com/grpc/grpc-dotnet/blob/eb813a798aa3c1009de65dc7e2c490106abd1401/test/Grpc.Net.Client.Tests/DiagnosticsTests.cs#L38-L45